src/app/modules/submission/email-submission/email-submission.controller.ts

```diff
@@ -30,267 +33,247 @@ const logger = createLoggerWithLabel(module)
 
 export const handleEmailSubmission: RequestHandler<
   { formId: string },
-  { message: string; submissionId?: string; spcpSubmissionFailure?: boolean },
+  { message: string; submissionId?: string; spcpSubmissionFailure?: true },
   { responses: FieldResponse[]; isPreview: boolean },
   { captchaResponse?: unknown }
 > = async (req, res) => {
   const { formId } = req.params
+  const attachments = mapAttachmentsFromResponses(req.body.responses)
+  let spcpSubmissionFailure: undefined | true
+
+  // Inlined utilities
   const logMeta = {
     action: 'handleEmailSubmission',
     ...createReqMeta(req as Request),
     formId,
   }
-  // Retrieve form
-  const formResult = await FormService.retrieveFullFormById(
-    formId,
-  ).andThen((form) => EmailSubmissionService.checkFormIsEmailMode(form))
-  if (formResult.isErr()) {
-    logger.error({
-      message: 'Error while retrieving form from database',
-      meta: logMeta,
-      error: formResult.error,
-    })
-    const { errorMessage, statusCode } = mapRouteError(formResult.error)
-    return res.status(statusCode).json({ message: errorMessage })
-  }
-  const form = formResult.value
 
-  // Check that form is public
-  const formPublicResult = FormService.isFormPublic(form)
-  if (formPublicResult.isErr()) {
-    logger.warn({
-      message: 'Attempt to submit non-public form',
-      meta: logMeta,
-      error: formPublicResult.error,
+  // Stores a predefined message and returns a function that can be used for mapErr
+  const logErrorWithMeta = (
```

Comment:
i think this is an overoptimization; we should just verbosely write out the logging. It makes it easier to reason about and aids when developers use the logger to check logs.

Reply:
^ 100% agree